### PR TITLE
CNI render support args

### PR DESCRIPTION
**Pull request by zirain**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/infra/plugin/plugin.go [hunk 1]
```diff
@@ -58,7 +58,22 @@ func RenderCNI(cluster *scope.Cluster) ([]byte, error) {
...
+	tpl, err := t.Funcs(sprig.TxtFuncMap()).Parse(out)
+	if err != nil {
+		return nil, err
+	}
+
+	var b bytes.Buffer
+	if err := tpl.Execute(&b, map[string]interface{}{
+		"ServiceCIDR": cluster.ServiceCIDR,
+		"PodCIDR":     cluster.PodCIDR,
+		"Extra":       cluster.CNIExtra,
```
Comment (hzxuzhonghu) [inline pkg/infra/plugin/plugin.go:71]: I can not find these three args in template, do you miss updating it

File: pkg/infra/scope/cluster_test.go [hunk 2]
```diff
@@ -17,14 +17,138 @@ limitations under the License.
...
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			input, err := readFromYaml(tc.name)
+			assert.NoError(t, err)
+
+			tc.expect.Cluster = input
+			assert.Equal(t, tc.expect, NewCluster(input))
+		})
```
Comment (hzxuzhonghu) [inline pkg/infra/scope/cluster_test.go:133]: not sure i understand what is the test coverage

Comment (zirain) [reply pkg/infra/scope/cluster_test.go:133]: make sure `CNIExtra` go the value as expected.

File: pkg/infra/plugin/plugin.go (hunk 1 above)
Comment (zirain) [reply pkg/infra/plugin/plugin.go]: refactor this, also add a UT for it.

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**